test/integration/consul-container/libs/cluster/container.go

```diff
@@ -156,6 +157,20 @@ func NewConsulContainer(ctx context.Context, config Config, cluster *Cluster, po
 
 		info AgentInfo
 	)
+	if utils.Debug {
+		for i := 0; i < 10; i++ {
```

Comment:
I guess we can replace this with `retry-go`, which is imported to the integ test:

https://github.com/hashicorp/consul/blob/44b39240a86bc94ddc67bc105286ab450bd869a9/test/integration/consul-container/libs/cluster/cluster.go#L141-L157

Reply:
Thank you for the review @huikang ! I changed that and also added few changes to print some data about the node when printing the debug port to help identify which container we would like to debug in a multi Consul test.